google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -67,13 +69,26 @@ def _set_enable_caching_value(
           Required. The dictionary of pipeline spec.
      enable_caching (bool):
           Required. Whether to enable caching.
+     cache_components (List[str]):
+          Required. List of component names to be cached.
     """
-    for component in [pipeline_spec["root"]] + list(
-        pipeline_spec["components"].values()
-    ):
-        if "dag" in component:
-            for task in component["dag"]["tasks"].values():
-                task["cachingOptions"] = {"enableCache": enable_caching}
+    if cache_components is None or enable_caching == False:
+        for component in [pipeline_spec["root"]] + list(
+            pipeline_spec["components"].values()
+        ):
+            if "dag" in component:
+                for task in component["dag"]["tasks"].values():
+                    task["cachingOptions"] = {"enableCache": enable_caching}
+    else:
+        for component in [pipeline_spec["root"]] + list(
+            pipeline_spec["components"].values()
+        ):
+            if "dag" in component:
+                for task_name, task in component["dag"]["tasks"].items():
+                    if task_name in cache_components:
+                        task["cachingOptions"] = {"enableCache": True}
+                    else:
+                        task["cachingOptions"] = {"enableCache": False}                        
```

Comment:
Same code was repeated here. Could you make it so we only use the same code once?

Reply:
Reduced the redundant code a bit, however cannot keep the inner `for` loop common because in one scenario we're using `values` method and in the other `items` method.
The way to deal with further logic is also distinct, and if we try to keep the conditional inside the `for` loop, it will use up additional compute each time we iterate through the loop, checking the same condition which essentially won't change.
Do have a look, and let me know if there are any further requirements/changes.